qmcpy/true_measure/matern.py

```diff
@@ -0,0 +1,89 @@
+            0.95042189]])
+
+    References:
+
+        [1] scikit-learn developers, Matern kernel. 
+        https://scikit-learn.org/stable/modules/generated/sklearn.gaussian_process.kernels.Matern.html.
+        Accessed 2023.
+    """
+    def __init__(self, sampler, x_values, length_scale = 1.0, nu = 1.5, variance = 1.0, mean = [], decomp_type='PCA'):
+        """
+        Matern kernel: calculates covariance over a metric space based only on the distance between points.
+        More information can be found at [1].
+        
+        Args:
+            sampler (DiscreteDistribution/TrueMeasure): A 
+                discrete distribution from which to transform samples or a
+                true measure by which to compose a transform. 
+            x_values (array): The positions of points on a metric space.
+            length_scale (float): Determines "peakiness", or how correlated 
+                two points are based on their distance.
+            nu (float): The "smoothness" of the Matern function. e.g. nu=1.5
+                implies a once-differentiable function, while nu=2.5 implies twice
+                differentiability. Meanwhile, when nu=0.5, the Matern kernel equals
+                the RBF kernel, while when nu=inf, it equals the absolute exponential
+                kernel. Note that nu values not in [0.5, 1.5, 2.5, inf] will be 
+                ~10x slower to run.
+            variance (float): The variance (or the diagonal elements of the
+                covariance matrix) at each point. 
+            mean (float): mu for Normal(mu,sigma^2)
+            decomp_type (str): method of decomposition either  
+                "PCA" for principal component analysis or 
+                "Cholesky" for cholesky decomposition.
+        """
+        #assert len(x_values) == len(mean)
+        if not (len(x_values) == len(mean)):
+            raise DimensionError("The dimensions of the position array and means must be equal.")
+        if not (isinstance(sampler, DiscreteDistribution) or isinstance(sampler, TrueMeasure)):
+            raise ParameterError("sampler input should either be a DiscreteDistribution or TrueMeasure.")
+        self.x_values = x_values
+        self.length_scale = length_scale
+        self.nu = nu
+        self.variance = variance
+        kernel = gp.kernels.Matern(length_scale = length_scale, nu=nu)
+        tnp = np.array([x_values]).T
+        #print(tnp)
+        self.covariance = variance * kernel.__call__(tnp) #takes array of size (# samples) x (# data points)
+        #print(covariances)
+        sampler_out = sampler.spawn(1, len(x_values))[0] #new version of sampler
+        super().__init__(sampler_out, mean=mean, covariance=self.covariance, decomp_type=decomp_type)
+
+    def _spawn(self, sampler):
+        return Matern(sampler.spawn(1, len(self.x_values))[0], self.x_values, length_scale=self.length_scale, nu=self.nu, variance=self.variance, 
```

Comment:
the sampler passed into this function is already a spawn, no need to spawn another one